doc/design/externally-manage-pf.md

```diff
@@ -0,0 +1,212 @@
+// NicSelector selects the NICs to be configured
+NicSelector SriovNetworkNicSelector `json:"nicSelector"`
+// +kubebuilder:validation:Enum=netdevice;vfio-pci
+// The driver type for configured VFs. Allowed value "netdevice", "vfio-pci". Defaults to netdevice.
+DeviceType string `json:"deviceType,omitempty"`
+// RDMA mode. Defaults to false.
+IsRdma bool `json:"isRdma,omitempty"`
+// mount vhost-net device. Defaults to false.
+NeedVhostNet bool `json:"needVhostNet,omitempty"`
+// +kubebuilder:validation:Enum=eth;ETH;ib;IB
+// NIC Link Type. Allowed value "eth", "ETH", "ib", and "IB".
+LinkType string `json:"linkType,omitempty"`
+// +kubebuilder:validation:Enum=legacy;switchdev
+// NIC Device Mode. Allowed value "legacy","switchdev".
+EswitchMode string `json:"eSwitchMode,omitempty"`
+// +kubebuilder:validation:Enum=virtio
+// VDPA device type. Allowed value "virtio"
+VdpaType string `json:"vdpaType,omitempty"`
+// Exclude device's NUMA node when advertising this resource by SRIOV network device plugin. Default to false.
+ExcludeTopology bool `json:"excludeTopology,omitempty"`
++ // don't create the virtual function only assign to the driver and allocated them to device plugin. Defaults to false.
++ ExternallyManaged bool `json:"externallyManaged,omitempty"`
+}
+```
+
+For SriovNetworkNodeState
+
+```golang
+type Interface struct {
+PciAddress        string    `json:"pciAddress"`
+NumVfs            int       `json:"numVfs,omitempty"`
+Mtu               int       `json:"mtu,omitempty"`
+Name              string    `json:"name,omitempty"`
+LinkType          string    `json:"linkType,omitempty"`
+EswitchMode       string    `json:"eSwitchMode,omitempty"`
+VfGroups          []VfGroup `json:"vfGroups,omitempty"`
++ ExternallyManaged bool      `json:"externallyManaged,omitempty"`
+}
+```
+
+### Implementation Details/Notes/Constraints
+
+#### Webhook
+For the webhook we add more validations when the policy contains `ExternallyManaged: true`
+* `numVfs` in the policy equal is equal or lower the number of virtual functions on the system
+* `MTU` in the policy equals the MTU we discover on the PF
+
+#### Controller/Manager
+
+The changes in the manager for this feature are minimal we only copy the `ExternallyManaged` boolean from the policy
+to the generated `nodeState.Spec`
+
+#### Config Daemon
+
+This is where most of the changes for this feature are implemented.
+
+First step we will do a validation same as on the webhook to check the PF have everything we need to apply the requested
+policy, by checking the `numVfs` and `MTU`.
+Next config-daemon will skip all the PF configuration like `numVfs` and `MTU`. he will only preform the virtual function 
+driver allocation, administrative mac allocation and MTU. Last step as always will be to reset the device plugin so
+kubelet will be able to discover the SR-IOV devices.
+
+The config-daemon will also save on the node a cache of the last applied policy. this is needed to be able and understand
```

Comment:
note: we already write a checkpoint file on daemon start see:

`see: NodeStateStatusWriter.writeCheckpointFile()`

perhaps this can be reused for last applied state

Reply:
that file is only for virtual env and it save something else from what we need